notebooks/concepts/imviz_subsampled_histogram.ipynb

```diff
@@ -0,0 +1,174 @@
+    "\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2e243795-8ed3-4410-99f1-4706f9fbb1b2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#shape = (800, 800)\n",
+    "#shape = (1600, 1600)\n",
+    "shape = (2000, 2000)\n",
+    "n_sources = 100\n",
+    "param_ranges = {'amplitude': [500, 1000],\n",
+    "                'x_mean': [0, shape[1]],\n",
+    "                'y_mean': [0, shape[0]],\n",
+    "                'x_stddev': [1, 5],\n",
+    "                'y_stddev': [1, 5],\n",
+    "                'theta': [0, np.pi]}\n",
+    "sources = make_random_gaussians_table(n_sources, param_ranges, seed=42)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "17a6b41a-a4fc-4d53-9254-5cad1cd0e717",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "arr = make_gaussian_sources_image(shape, sources) + 5.0\n",
+    "\n",
+    "# Noise\n",
+    "rng = np.random.RandomState(12345)\n",
+    "arr += rng.normal(loc=0.0, scale=2.0, size=shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9121966d-0b1c-4d28-b538-dc85fabff69a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fill = 666\n",
+    "arr[:4, :] = fill\n",
+    "arr[-4:, :] = fill\n",
+    "arr[:, :4] = fill\n",
+    "arr[:, -4:] = fill"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "165ea850-70f9-4033-8d87-88823973940e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xstep = max(1, round(arr.shape[1] / 400))\n",
+    "ystep = max(1, round(arr.shape[0] / 400))\n",
+    "subarr = arr[::ystep, ::xstep]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "210250bc-c782-47bd-b6fc-fefc053d6cc3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(xstep, ystep)\n",
+    "print(arr.shape, subarr.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9849ddf0-99c9-418b-866f-ee74ec248fa0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.imshow(arr)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ecad40a5-a6a4-404f-b7bd-754e9876f645",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.imshow(subarr)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "90a5b346-105a-480e-aed2-8485bdcad936",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig, ax = plt.subplots()\n",
+    "_ = ax.hist(arr, bins=10)\n",
+    "_ = ax.set_title(\"Original\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c8984ab-4646-40e5-9dca-d01fd03daa08",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fig2, ax2 = plt.subplots()\n",
+    "_ = ax2.hist(subarr, bins=10)\n",
+    "_ = ax2.set_title(\"Gridded\")"
```

Comment:
Okay, I think I see what @camipacifici is saying about the "frame" being wiped out. But then again, are the percentile vmin/vmax calculated using all the data, randomly sampled data, or gridded downsampled data in Imviz? Maybe @astrofrog knows.

<table>
<tr>
<td>
<img width="524" alt="Screenshot 2024-02-29 140608" src="https://github.com/spacetelescope/jdaviz/assets/2090236/0ce18680-a989-4be2-84fd-95862a8882e0">
</td>
<td>
<img width="524" alt="Screenshot 2024-02-29 140657" src="https://github.com/spacetelescope/jdaviz/assets/2090236/2cf3a8b7-d185-4214-a680-dc1ea54b1650">
</td>
</tr>
</table>

Reply:
I am not sure if the cons of gridding matters in the end if it is the vmin/vmax that you are worried about. This is what Plot Options actually look like with the framed data in this notebook:

| No gridding (main) | With gridding (this PR) |
| --- | --- |
| <img width="304" alt="hist_original" src="https://github.com/spacetelescope/jdaviz/assets/2090236/4ddb12d8-a263-4aa6-9280-259bf0d7605f"> | <img width="315" alt="hist_downsampled" src="https://github.com/spacetelescope/jdaviz/assets/2090236/5e579b60-bafa-4b18-98a2-2462727fda69"> |
| <img width="212" alt="vminmax_original" src="https://github.com/spacetelescope/jdaviz/assets/2090236/29341e68-7f26-4960-9246-d4064b5ec492"> |  <img width="163" alt="vminmax_downsampled" src="https://github.com/spacetelescope/jdaviz/assets/2090236/edea84fd-ace3-4b86-8c70-f39a0d9bf141"> |

Reply:
Here are the Imviz vmin/vmax when I changed the fill value to 99999 as Cami suggested: 🤷‍♀️ 

<table>
<tr>
<td>
<img width="161" alt="Screenshot 2024-02-29 170435" src="https://github.com/spacetelescope/jdaviz/assets/2090236/80fc0cdc-7477-460e-bde8-b509b9e0ea70">
</td>
<td>
<img width="176" alt="Screenshot 2024-02-29 170237" src="https://github.com/spacetelescope/jdaviz/assets/2090236/2364c70b-2b38-47c6-88b3-0483d397a7bb">
</td>
</tr>
</table>

Reply:
To repeat what I said on Slack here: As things currently stands on `main`, gridding improves performance and does not seem to affect significantly how vmin/vmax is computed. If the real goal is to accurately have histogram and vmin/vmax reflect the real (full) data via true random sampling, I think the fix is actually in glue-core and would be quite involved. (Though my understanding of glue internals is very limited so feel free to correct me.)